doc/manual/src/command-ref/nix-collect-garbage.md

```diff
@@ -1,24 +1,49 @@
 # Name
 
-`nix-collect-garbage` - delete unreachable store paths
+`nix-collect-garbage` - delete unreachable [store objects]
 
 # Synopsis
 
 `nix-collect-garbage` [`--delete-old`] [`-d`] [`--delete-older-than` *period*] [`--max-freed` *bytes*] [`--dry-run`]
 
 # Description
 
-The command `nix-collect-garbage` is mostly an alias of [`nix-store
---gc`](@docroot@/command-ref/nix-store/gc.md), that is, it deletes all
-unreachable paths in the Nix store to clean up your system. However,
-it provides two additional options: `-d` (`--delete-old`), which
-deletes all old generations of all profiles in `/nix/var/nix/profiles`
-by invoking `nix-env --delete-generations old` on all profiles (of
-course, this makes rollbacks to previous configurations impossible);
-and `--delete-older-than` *period*, where period is a value such as
-`30d`, which deletes all generations older than the specified number
-of days in all profiles in `/nix/var/nix/profiles` (except for the
-generations that were active at that point in time).
+The command `nix-collect-garbage` is mostly an alias of [`nix-store --gc`](@docroot@/command-ref/nix-store/gc.md),
+that is, `nix-collect-garbage` deletes all unreachable [store objects] in the Nix store to clean up your system.
+
+However, it provides two additional options which also delete old [profiles], allowing potentially more [store objects] to be deleted because profiles are also garbage collection roots.
+These options are the equivalent of running variations of [`nix-env --delete-generations`](@docroot@/command-ref/nix-env/delete-generations.md)
+on various profiles, prior to running `nix-collect-garbage` (or just `nix-store --gc`) without any flags.
+
+Do note that deleting previous configurations makes rollbacks to them impossible.
+Because these flags potentially delete older configurations used by other users on the system, these flags should be used with care.
+
+## Locations searched for profiles
+
+`nix-collect-garbage` cannot know about all profiles; that information doesn't exist.
+Instead, it looks looks in a few locations for profiles, and acts on all profiles it finds there.
+
+Firstly, `nix-collect-garbage` looks in the default profile locations as specified in the [profiles] section of the manual.
+
+Secondly, `nix-collect-garbage` looks in `$NIX_STATE_DIR/profiles` and `$NIX_STATE_DIR/profiles/per-user`.
+With the exception of `$NIX_STATE_DIR/profiles/per-user/root` and `$NIX_STATE_DIR/profiles/default`, members of these directories are no longer used by default.
+`nix-collect-garbage` look there anyways in order to clean up profiles from older versions of Nix.
```

Comment:
I'm very much not keen on documenting that. These locations aren't something that people should care about and exposing them just increases our API surface for no good reason. People relying on `/nix/var/nix/profiles` is already what caused most of the troubles with https://github.com/NixOS/nix/pull/5226 (and similarly, https://github.com/NixOS/nix/pull/5588 is blocked by the fact that we expose more files than we should)

Reply:
We still have to answer the general question how to deal with such issues: There are some things that are accessible, i.e. are already part of the API, but should not be relied on.

Reply:
I added a note, but not marking this resolve yet until you both approve.

Reply:
Looks good, thanks :+1: